draft-ietf-core-dns-over-coap.md

```diff
@@ -182,16 +182,34 @@ or automatic configuration, e.g., using a CoRE resource directory
 {{-ddr}}.
 Automatic configuration SHOULD only be done from a trusted source.
 
-Support for SVCB Resource Records {{-svcb}}, {{-svcb-dns}} or DNR Service Parameters {{-dnr}}
-are not specified in this document.
-{{-core-dnr}} explores solutions for CoAP for these mechanisms.
-
 When discovering the DNS resource through a link mechanism that allows describing a resource type
 (e.g., the Resource Type Attribute in {{-core-link-format}}), the resource type "core.dns" can be
 used to identify a generic DNS resolver that is available to the client.
 
-While there is no path specified it is RECOMMENDED to use the root path "/" for the DNS resource to
-keep the CoAP requests small.
+A DoC server can also be discovered using SVCB Resource Records (RR) {{-svcb}}, {{-svcb-dns}} or DNR
+Service Parameters {{-dnr}}.
+\[TBD: draft-lenders-core-coap-dtls-svcb\] provides solutions
+to discover CoAP over (D)TLS servers using the "alpn" SvcParam. This document specifies "docpath" as
+a single-valued SvcParamKey whose value MUST be a CBOR sequence of 0 or more text strings (see
+{{-cbor}}), delimited by length (in total octets) for the the SvcParamValue field. If the
+SvcParamValue ends within a CBOR text string, the SVCB RR MUST be considered as malformed.
+
+Note, that this specifically does not surround the text string sequence with a CBOR array or similar
+CBOR data item. This path format was chosen to coincide with the path representation in CRIs
+({{-cri}}). Furthermore, it is easily transferable into a sequence of CoAP Uri-Path options by
+mapping the initial byte of any present CBOR text string (see {{-cbor, Section 3}}) into the Option
+Delta and Option Length of the CoAP option, provided these CBOR text strings are all of a length
+between 0 and 12 octets (see {{-coap, Section 3.1}}). Likewise, it can be transfered into a URI
+path-abempty form (see {{-uri, Section 3.3}}) by replacing the initial byte of any present CBOR text
+string with the "/" character, provided these CBOR text strings are all of a length lesser than 24
+octets.
```

Comment:
(Could also be value-list because that's what zone files already use)

Reply:
See https://github.com/core-wg/draft-dns-over-coap/pull/28/files#diff-3447d898853b49b9d631406ac996197ef45750f12dbd2bc394787a3ac56933bbR196-R197 for now. We can also use the value list route, but need to look into that format first.

Reply:
value-list is described at https://www.rfc-editor.org/rfc/rfc9460.html#name-decoding-a-comma-separated -- not sure where that comes from originally or how that language works as a whole.

Reply:
Mh... I think I prefer CBOR-diag, if that is how the value-list format works. Your [off-line] worries about commas seem to me to be unjustified when looking at the value-list format.

Reply:
> "docpath" sounds a bit vague - could it be "coappath" or "coapath" ?

How is "coappath" less vague (referring to the transport protocol) than "docpath" (referring to DoC, DNS over CoAP, the application protocol used over said transport)? The name was chosen in line with "dohpath" which uses an URI template which we don't need for DoC. There a simple path suffices.

Reply:
oh - thinking about it more you are more than correct.  I do not know what came over me.

Reply:
No worries. We were actually wondering off-line, if there was some insight with the history of "dohpath" in that and if generalization would indeed make sense. 

Would it have made sense if e.g. "dohpath" would have been called "httppath", too? The fact that we are asking for a DNS resource can be inferred from the query or just the fact that they are in a DNR-related message. As such, having the information DNS+path encoded in this codepoint might be redundant and just having the path information would suffice and even re-usable for other use cases in the future.

Moreover, as pointed out, differing from "dohpath" we only encode a path in the value (not a URI template). So it _is_ already more generalized.

Reply:
Summarizing my original question: Seems to be OK, https://www.rfc-editor.org/rfc/rfc9460.html#name-decoding-a-comma-separated- has an example of a quoted / escaped SVCB parame value, so that part is fine. (and IIUC, dohpath is also fine from the last posts)